app/src/main/java/ch/hikemate/app/ui/map/HikeDetailsScreen.kt

```diff
@@ -188,95 +268,44 @@ fun HikeDetailScreen(
 
   // Show the selected hike on the map
   // OnLineClick does nothing, the line should not be clickable
-  val hikeLineColor = route.getColor()
-  Log.d("HikeDetailScreen", "Drawing hike on map: ${route.bounds}")
-  MapUtils.showHikeOnMap(mapView = mapView, hike = route, color = hikeLineColor, onLineClick = {})
-
-  LaunchedEffect(Unit) {
-    if (authViewModel.currentUser.value == null) {
-      Log.e("HikeDetailScreen", "User is not signed in")
-      return@LaunchedEffect
-    }
-    profileViewModel.getProfileById(authViewModel.currentUser.value!!.uid)
-  }
-
-  val errorMessageIdState = profileViewModel.errorMessageId.collectAsState()
-  val profileState = profileViewModel.profile.collectAsState()
-
-  AsyncStateHandler(
-      errorMessageIdState = errorMessageIdState,
-      actionContentDescriptionStringId = R.string.go_back,
-      actionOnErrorAction = { navigationActions.navigateTo(Route.MAP) },
-      valueState = profileState,
-  ) { profile ->
-    Box(modifier = Modifier.fillMaxSize().testTag(Screen.HIKE_DETAILS)) {
-      // Map
-      AndroidView(
-          factory = { mapView },
-          modifier =
-              Modifier.fillMaxWidth()
-                  .padding(bottom = 300.dp) // Reserve space for the scaffold at the bottom
-                  .testTag(TEST_TAG_MAP))
-      // Back Button at the top of the screen
-      BackButton(
-          navigationActions = navigationActions,
-          modifier = Modifier.padding(start = 16.dp, end = 16.dp).safeDrawingPadding(),
-          onClick = { listOfHikeRoutesViewModel.clearSelectedRoute() })
-      // Zoom buttons at the bottom right of the screen
-      ZoomMapButton(
-          onZoomIn = { mapView.controller.zoomIn() },
-          onZoomOut = { mapView.controller.zoomOut() },
-          modifier =
-              Modifier.align(Alignment.BottomEnd)
-                  .padding(bottom = MapScreen.BOTTOM_SHEET_SCAFFOLD_MID_HEIGHT + 8.dp))
-
-      // Hike Details bottom sheet
-      HikeDetails(
-          detailedRoute = detailedRoute,
-          savedHikesViewModel = savedHikesViewModel,
-          elevationData = elevationData,
-          userHikingLevel = profile.hikingLevel,
-          onRunThisHike = { navigationActions.navigateTo(Screen.RUN_HIKE) },
-      )
-    }
-  }
+  val hikeLineColor = hike.color
```

Comment:
Nitpick: I think you could put `hike.color` directly as the parameter. This is longer 😂

Reply:
Why did I do this in the first place 😂